pilot/pkg/networking/core/configgen.go

```diff
@@ -39,8 +39,7 @@ type ConfigGenerator interface {
 
 	// BuildDeltaClusters returns both a list of resources that need to be pushed for a given proxy and a list of resources
 	// that have been deleted and should be removed from a given proxy. This is Delta CDS output.
-	BuildDeltaClusters(node *model.Proxy, push *model.PushContext, updates *model.PushRequest,
-		watched *model.WatchedResource) ([]*discovery.Resource, []string, model.XdsLogDetails, bool)
+	BuildDeltaClusters(node *model.Proxy, updates *model.PushRequest, watched *model.WatchedResource) ([]*discovery.Resource, []string, model.XdsLogDetails)
```

Comment:
We need to return whether or not delta was used form `BuildDeltaClusters`, as was done in the inital PR. We do not know if delta was actually used, generation may have fallen back on sotw

Reply:
Ack. I am thinking about what is the best way to do it. Will make that change.